### PR TITLE
Deprecating selection

### PR DESCRIPTION
**Pull request by ChristopherDavisUCI**
As requested by @mattijn here: https://github.com/altair-viz/altair/pull/2908#issuecomment-1440698123

Add a deprecation decorator to `selection` and update the usage in a few spots.

### CODE REVIEW

File: altair/vegalite/v5/api.py [hunk 1]
```diff
@@ -418,6 +418,10 @@ def _selection(type=Undefined, **kwds):
     return param(select=select, **param_kwds)
 
 
+@utils.deprecation.deprecated(
+    message="""'selection' is deprecated.
+    Although it is possible to create a point selection using 'selection(type="point")' and an interval selection using 'selection(type="interval")', for consistency, it is recommended to use 'selection_point()' or 'selection_interval()' instead."""
```
Comment (joelostblom) [inline altair/vegalite/v5/api.py:423]:
```suggestion
   Use 'selection_point()' or 'selection_interval()' instead; these functions also include more helpful docstrings."""
```

**Review by joelostblom — CHANGES_REQUESTED** [review]
Look good! One minor suggestion.

Do you think we should follow the same path for `alt.binding` for consistency? https://joelostblom.github.io/altair-docs/user_guide/API.html#api-functions

**Comment by ChristopherDavisUCI** [comment]
Good catch @mattijn, I was searching for `type="interval"` instead of `type='interval'`.  Should we always use one or the other, at least for this little case?

**Comment by mattijn** [comment]
Regarding:

> Do you think we should follow the same path for `alt.binding` for consistency? https://joelostblom.github.io/altair-docs/user_guide/API.html#api-functions

No. Not yet at least. See eg https://github.com/altair-viz/altair/issues/2760, which uses `alt.binding(input='color')`

**Pull request merged**